src/get-started/timestamping-tutorial.md

```diff
@@ -0,0 +1,1017 @@
+  blockchain network.
+
+## Define Service
+
+Service is a group of templated transactions (we have defined them before).
+It has a name and a unique ID to determine the service inside the blockchain.
+We define our new service in [src/lib.rs][src/lib.rs]:
+
+```rust
+#[derive(Debug, Default)]
+pub struct Service;
+```
+
+To turn `Service` into a blockchain service, we need to implement the Exonum
+`Service` trait for it.
+
+!!!tip
+   Read more on how to turn a type into a blockchain service in the
+   [Interface with Exonum Framework](../architecture/services.md#interface-with-exonum-framework)
+   section.
+
+```rust
+impl blockchain::Service for Service {
+    fn service_id(&self) -> u16 {
+        TIMESTAMPING_SERVICE
+    }
+
+    fn service_name(&self) -> &'static str {
+        SERVICE_NAME
+    }
+
+    fn state_hash(&self, view: &dyn Snapshot) -> Vec<Hash> {
+        let schema = Schema::new(view);
+        schema.state_hash()
+    }
+
+    fn tx_from_raw(
+        &self,
+        raw: RawTransaction,
+    ) -> Result<Box<dyn Transaction>, StreamStructError> {
+        let tx = TimeTransactions::tx_from_raw(raw)?;
+        Ok(tx.into())
+    }
+
+    fn wire_api(&self, builder: &mut ServiceApiBuilder) {
+        PublicApi::wire(builder);
+    }
+}
+
+```
+
+The code above declares a number of methods:
+
+- `service_id` - returns the ID of the service
+- `service_name` - returns the name of the service
+- `state_hash` - calculates the hash of
+  [the blockchain state](../glossary.md#blockchain-state). The method
+  should return [a vector of hashes](../architecture/services.md#state-hash)
+  of the [Merkelized service tables](../glossary.md#merkelized-indices).
+- `tx_from_raw` - deserialization of transactions processed by the service.
+  If the incoming transaction is built successfully, we put it into
+  a `Box<_>`.
+- `wire_api` - connects the API endpoints of the service to the API of the
+  blockchain.
+
+## Create Service Instance
+
+We define the `ServiceFactory` structure in [src/lib.rs][src/lib.rs]. This
+structure builds an instance of our service.
+
+```rust
+#[derive(Debug, Clone, Copy)]
+pub struct ServiceFactory;
+```
+
+The `ServiceFactory` returns the name of the service and includes the
+`make_service` function which will later allow us to include the new service
+into the `NodeBuilder`.
+
+```rust
+impl fabric::ServiceFactory for ServiceFactory {
+    fn service_name(&self) -> &str {
+        SERVICE_NAME
+    }
+
+    fn make_service(
+        &mut self,
+        _: &fabric::Context,
+    ) -> Box<dyn blockchain::Service> {
+    }
+}
+```
+
+## Create Builder for Exonum Node With Timestamping
+
+As the final step of configuring the backend of our Timestamping Service, we
+add the [src/main.rs][src/main.rs] file, which lets us launch an Exonum node
+with additional services.
+
+In this file we import Exonum, its Configuration service, the Time oracle which
```

Comment:
Configuration Service
time oracle or Time Oracle. But please use the same (de)capitalization through the whole document